proteuslib/tools/parameter_sweep.py

```diff
@@ -278,5 +397,12 @@ def parameter_sweep(model, sweep_params, outputs, results_file, optimize_functio
 
         # Save the global data
         np.savetxt(results_file, save_data, header=data_header, delimiter=', ', fmt='%.6e')
+    else:
+        save_data = None
+
+    # Broadcast the results to all processors
+    save_data = comm.bcast(save_data, root=0)
```

Comment:
Similar to before, we should use `Bcast`; the non-zero ranks should know the size of `save_data` in advance.